### PR TITLE
Us 266729 model based on screen size and model size

### PR DESCRIPTION
**Pull request by DomanskiJakub**

### CODE REVIEW

File: android/core/src/main/java/com/pega/vehicledamagemodeling/LimitedCameraInputController.java [hunk 1]
```diff
@@ -111,4 +114,41 @@ private boolean isCrossingUpLimit(Vector3 rotateVector, float deltaYRotate) {
         tmpV2.rotate(rotateVector, deltaYRotate);
         return tmpV2.y < ROTATE_UP_LIMIT;
     }
+
+    public void setUpPosition(BoundingBox boundingBox, int width, int height) {
+        float distanceModelToCamera = Math.max(boundingBox.getDepth(), boundingBox.getWidth());
+        float xPositionByPythagorasTriangle = distanceModelToCamera * 12 / 13;
```
Comment (sanchin) [inline android/core/src/main/java/com/pega/vehicledamagemodeling/LimitedCameraInputController.java]: What are those magic numbers? Where do they come form? Are they model specific? If so, they should be at least extracted as a meaningful constant, that allows for a quick change in one place. 
For example - is this 13 the same as the one below? If I need to change the first one should I also change the second one?

File: android/core/src/main/java/com/pega/vehicledamagemodeling/LimitedCameraInputController.java [hunk 2]
```diff
@@ -111,4 +114,41 @@ private boolean isCrossingUpLimit(Vector3 rotateVector, float deltaYRotate) {
         tmpV2.rotate(rotateVector, deltaYRotate);
         return tmpV2.y < ROTATE_UP_LIMIT;
     }
+
+    public void setUpPosition(BoundingBox boundingBox, int width, int height) {
```
Comment (sanchin) [inline android/core/src/main/java/com/pega/vehicledamagemodeling/LimitedCameraInputController.java]: Width and height params are not used

File: android/core/src/main/java/com/pega/vehicledamagemodeling/LimitedCameraInputController.java [hunk 3]
```diff
@@ -111,4 +114,41 @@ private boolean isCrossingUpLimit(Vector3 rotateVector, float deltaYRotate) {
...
+    public float getMatchingFieldOfView(int width, int height) {
+        int min = Math.min(width, height);
+        int max = Math.max(width, height);
+        float div = (float) min / max;
+        float abs = 1f - Math.abs(div - 1f);
+        if (isStartOrientationVertical) {
+            if (isCurrentOrientationVertical(width, height)) {
+                return 90;
+            } else {
+                return 40 + abs * 50;
```
Comment (sanchin) [inline android/core/src/main/java/com/pega/vehicledamagemodeling/LimitedCameraInputController.java]: Again - magic number. What's the logic behind them?

File: android/core/src/main/java/com/pega/vehicledamagemodeling/api/PartSelectionDetector.java [hunk 4]
```diff
@@ -53,7 +53,7 @@ public boolean touchDown (int screenX, int screenY, int pointer, int button) {
     public boolean touchUp (int screenX, int screenY, int pointer, int button) {
         if (screenTouchedMarker.epsilonEquals(screenX, screenY, SENSITIVITY)) {
             ModelInstance selectedPart = getSelectedPartId(screenX, screenY);
-            if (selectedPart != null) {
+            if (selectedPart != null && !selectedPart.materials.get(0).id.equals("landing-gear")) {
```
Comment (sanchin) [inline android/core/src/main/java/com/pega/vehicledamagemodeling/api/PartSelectionDetector.java]: landing-gear? In a car?! This is also very model-specific, should be extracted, removed, or at least commented what's it for.

**Review by sanchin — CHANGES_REQUESTED** [review]

File: android/core/src/main/java/com/pega/vehicledamagemodeling/api/PartSelectionDetector.java (hunk 4 above)
Comment (Jarek-SRT) [reply android/core/src/main/java/com/pega/vehicledamagemodeling/api/PartSelectionDetector.java]: Hardcoding a single invalid id which happened to be present in the current model is not a good idea.  A better check would be to verify if the id is on the list of valid ids listed in JSON schema.

**Comment by lapwingg** [comment]
I reverted to prior version in file VehicleDamageModelingActivity.java.  Ultimately, I didn't need it when I was creating Custom Module.

File: android/core/src/main/java/com/pega/vehicledamagemodeling/api/PartSelectionDetector.java [hunk 5]
```diff
@@ -53,14 +53,16 @@ public boolean touchDown (int screenX, int screenY, int pointer, int button) {
...
-            if (selectedPart != null) {
+            if (selectedPart != null && !selectionService.isExcludedPart(selectedPart)) {
                 uiUpdateCallback.enableCheckButton();
                 selectionService.setSelectedPart(selectedPart);
             }
         }
         return false;
     }
 
+
```
Comment (KarolinaSzypura) [inline android/core/src/main/java/com/pega/vehicledamagemodeling/api/PartSelectionDetector.java]: puste linie

File: android/core/src/main/java/com/pega/vehicledamagemodeling/api/SelectionService.java [hunk 6]
```diff
@@ -23,6 +23,7 @@
     private SelectedPartsRepository selectedPartsRepository;
     private Parser parser;
 
+
```
Comment (KarolinaSzypura) [inline android/core/src/main/java/com/pega/vehicledamagemodeling/api/SelectionService.java]: pusta linia

File: android/core/src/test/java/com/pega/vehicledamagemodeling/api/PartSelectionDetectorTest.java [hunk 7]
```diff
@@ -78,4 +79,26 @@ public void whenTapThenDetectAction() {
...
+        Camera camera = mock(PerspectiveCamera.class);
+        Ray ray = new Ray(new Vector3(0, -2, 0), new Vector3(0, 2, 0));
+        when(camera.getPickRay(screenX, screenY)).thenReturn(ray);
+        Array<ModelInstance> modelInstances = new Array<>();
+        ModelInstance modelInstance = mock(ModelInstance.class);
+        BoundingBox boundingBox = new BoundingBox(new Vector3(-1, -1, -1), new Vector3(1, 1, 1));
+        when(modelInstance.calculateBoundingBox(any(BoundingBox.class))).thenReturn(boundingBox);
+        modelInstances.add(modelInstance);
+        when(selectionService.isExcludedPart(modelInstance)).thenReturn(true);
+        PartSelectionDetector selectedDetector = new PartSelectionDetector(camera, modelInstances, selectionService, mock(UIUpdateCallback.class));
```
Comment (KarolinaSzypura) [inline android/core/src/test/java/com/pega/vehicledamagemodeling/api/PartSelectionDetectorTest.java]: zbyt długa linia, przydałoby się ją złamać
kilka innych w tej klasie też takich jest

File: android/core/src/main/java/com/pega/vehicledamagemodeling/LimitedCameraInputController.java [hunk 8]
```diff
@@ -111,4 +121,49 @@ private boolean isCrossingUpLimit(Vector3 rotateVector, float deltaYRotate) {
         tmpV2.rotate(rotateVector, deltaYRotate);
         return tmpV2.y < ROTATE_UP_LIMIT;
     }
+
+    public void setUpPosition(BoundingBox boundingBox) {
+        float distanceModelToCamera = Math.max(boundingBox.getDepth(), boundingBox.getWidth());
+        float xPositionByPythagorasTriangle = distanceModelToCamera * PYTHAGORAS_TRIANGLE_5_12_13_B / PYTHAGORAS_TRIANGLE_5_12_13_C;
```
Comment (KarolinaSzypura) [inline android/core/src/main/java/com/pega/vehicledamagemodeling/LimitedCameraInputController.java]: zbyt długa linia. Przydałoby się ją złamać

**Review by KarolinaSzypura — APPROVED** [review]

File: android/core/src/main/java/com/pega/vehicledamagemodeling/api/SelectionService.java [hunk 9]
```diff
@@ -17,11 +17,21 @@
 import com.badlogic.gdx.graphics.g3d.Material;
 import com.badlogic.gdx.graphics.g3d.ModelInstance;
 import com.badlogic.gdx.utils.Array;
+import com.google.gson.JsonArray;
 import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
```
Comment (KarolinaSzypura) [inline android/core/src/main/java/com/pega/vehicledamagemodeling/api/SelectionService.java]: linia do usunięcia

**Review by sanchin — APPROVED** [review]

File: android/core/src/main/java/com/pega/vehicledamagemodeling/api/PartSelectionDetector.java (hunk 4 above)
Comment (MartinJurczyk) [reply android/core/src/main/java/com/pega/vehicledamagemodeling/api/PartSelectionDetector.java]: changed

**Review by MartinJurczyk — APPROVED** [review]

**Pull request merged**